flutter-idea/src/io/flutter/run/daemon/DevToolsServerTask.java

```diff
@@ -0,0 +1,399 @@
+
+import com.google.common.collect.ImmutableList;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+import com.intellij.execution.ExecutionException;
+import com.intellij.execution.configurations.GeneralCommandLine;
+import com.intellij.execution.process.ProcessAdapter;
+import com.intellij.execution.process.ProcessEvent;
+import com.intellij.execution.process.ProcessHandler;
+import com.intellij.notification.Notification;
+import com.intellij.notification.NotificationType;
+import com.intellij.notification.Notifications;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.progress.ProcessCanceledException;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.project.ProjectManager;
+import com.intellij.openapi.project.ProjectManagerListener;
+import com.intellij.openapi.util.Key;
+import com.intellij.openapi.util.Version;
+import com.intellij.openapi.util.io.FileUtil;
+import com.intellij.openapi.util.registry.Registry;
+import com.jetbrains.lang.dart.ide.devtools.DartDevToolsService;
+import com.jetbrains.lang.dart.ide.toolingDaemon.DartToolingDaemonService;
+import com.jetbrains.lang.dart.sdk.DartSdk;
+import io.flutter.FlutterMessages;
+import io.flutter.FlutterUtils;
+import io.flutter.bazel.Workspace;
+import io.flutter.bazel.WorkspaceCache;
+import io.flutter.dart.DtdUtils;
+import io.flutter.sdk.FlutterSdk;
+import io.flutter.sdk.FlutterSdkUtil;
+import io.flutter.utils.JsonUtils;
+import io.flutter.utils.MostlySilentColoredProcessHandler;
+import io.flutter.utils.OpenApiUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicReference;
+
+class DevToolsServerTask extends Task.Backgroundable {
+  private static final Logger LOG = Logger.getInstance(DevToolsServerTask.class);
+  public static final String LOCAL_DEVTOOLS_DIR = "flutter.local.devtools.dir";
+  public static final String LOCAL_DEVTOOLS_ARGS = "flutter.local.devtools.args";
+  private @NotNull Project project;
+  private final AtomicReference<CompletableFuture<DevToolsInstance>> devToolsFutureRef;
+
+  private DaemonApi daemonApi;
+  private ProcessHandler process;
+
+  public DevToolsServerTask(@NotNull Project project,
+                            @NotNull String title,
+                            AtomicReference<CompletableFuture<DevToolsInstance>> devToolsFutureRef) {
```

Comment:
Is this certain to be non-null? If so, could you annotate it `@NotNull` and consider the same for `devToolsFutureRef`?